snippets/memoize.md

```diff
@@ -2,13 +2,16 @@
 
 Returns the memoized (cached) function.
 
-Use `Object.create(null)` to create an empty object without `Object.prototype` (so that those properties are not resolved if the input value is something like `'hasOwnProperty'`).
+Create an empty cache by instantiating a new `Map` object.
 Return a function which takes a single argument to be supplied to the memoized function by first checking if the function's output for that specific input value is already cached, or store and return it if not.
+Allow access to the `cache` by setting it as a property on the returned function.
 
 ```js
 const memoize = fn => {
-  const cache = Object.create(null);
-  return value => cache[value] || (cache[value] = fn(value));
+  const cache = new Map();
+  const cached = val => cache.has(val) ? cache.get(val) : (cache.set(val, fn(val)) && cache.get(val));
+  cached.cache = cache;
```

Comment:
Why are we exposing the Cache? This leads to outside interference leading to unpredictable results.. We should keep it a private should we not?